pkg/scheduler/scheduler_test.go

```diff
@@ -277,6 +277,72 @@ func TestSchedule(t *testing.T) {
 
 		wantSkippedPreemptions map[string]int
 	}{
+		"use second flavor when the first has no preemption candidates; WhenCanPreempt: Preempt": {
```

Comment:
This case can be considered as a sibling for https://github.com/kubernetes-sigs/kueue/blob/46749a22249ada4651b44ada661287b34702f7a4/pkg/scheduler/scheduler_test.go#L952

So, could you add a suffix `; WhenCanPreempt: TryNextFlavor` to the "not enough resources to borrow, fallback to next flavor"?